### PR TITLE
Sponsorship application detail view

### LINKED ISSUE #1728: Sponsors app: Create a view for Users associated with Sponsorships
Users associated with SponsorContacts should have a way to view the details of their Sponsorships that are in the 'APPLIED', 'APPROVED', and 'FINALIZED' state.

This may be accessible via the [`UserDetail`](https://github.com/python/pythondotorg/blob/2f5d5019ad89533cf863bd906515e35a21fa4bd1/users/views.py#L121) view and [associated template](https://github.com/python/pythondotorg/blob/master/templates/users/base.html) but doesn't necessarily need to be implemented in the `users` app.

Basic information that should be accessible for all associated `Sponsorship` objects:
  - Status (Applied, Approved, Finalized)
  - Effective and Expiration dates
  - Level and cost of sponsorship aggreement
  - Benefits
  - Associated contracts (Once #1702 is complete)

### LINKED ISSUE #1729: Refresh of UserDetail view
When navigating to "Your account" after signing in to www.python.org, a very bare bones "dashboard" is visible to the user.

With #1728 adding additional information/views that will likely be accessed via this area, we should take a look at reorganizing this area of the site and potentially making other improvements to rendering and styling.

At the very least, we should update navigation to be more intuitive, currently the sections are just kind of plopped in there. We should probably organize them in some hierarchy to allow for future expansion.

With the addition of sponsorship, something like:

* Your Account
    * View/Edit Profile
    * Change Password
* Membership
    * View/Edit PSF Membership
    * Nominations
* Sponsorship (if applicable
    * View Sponsorships

Where possible, we should reuse other navigation components and styling.

### PR DESCRIPTION
**Pull request by berinhard**
Fixes #1728
Fixes #1729 

@ewdurbin here are the screen shots:

Screenshot from 2021-02-17 13-15-50

Screenshot from 2021-02-17 13-15-57

### CODE REVIEW

File: templates/sponsors/sponsorship_detail.html [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    {% if sponsorship.finalized_on %}
+    <li>
+      <b>Finalized date:</b> {{ sponsorship.finalized_on }}
+    </li>
+    {% endif %}
+    <li>
+      <b>Level:</b> {{ sponsorship.level_name }}
+    </li>
+    <li>
+      <b>Agreed sponsorship fee:</b> ${{ sponsorship.sponsorship_fee|intcomma }}
```
Comment (ewdurbin) [inline templates/sponsors/sponsorship_detail.html]: After applying, this is currently showing the package price even if the benefits have been modified. We should show a similar "TBD" text as we do on the confirmation page.

File: static/sass/style.scss [hunk 2]
```diff
@@ -2055,12 +2055,45 @@ $colors: $blue, $psf, $yellow, $green, $purple, $red;
...
+      padding: 0.25em 1em;
+
+      &:hover {
+          .section-links {
+              display: inline;
+          }
+      }
+
+      span {
+        @extend %button-style-yellow;
```
Comment (ewdurbin) [inline static/sass/style.scss:2075]: Screen Shot 2021-04-13 at 3 44 23 PM
This button styling doesn't seem to be working for me locally.

**Comment by ewdurbin** [comment]
There are a couple of small things here that I think should be addressed, but the primary one is the styling of the navigation.

I am not sure why the styling shown in your screenshots isn't rendering for me, but if we could get something reliable and straight forward going, I think this is a good start.

**Comment by berinhard** [comment]
@ewdurbin I updated this PR with 2 new commits addressing your comments. About the buttons style, I didn't commit a few lines in the final `style.css` file (e0ddfbf) and now it should work for you.

About the sponsorship fee information, I added a condition (d1c70dd) to check if the price should be displayed, but I'm wondering if the condition isn't to restrictive. Because it makes sense to me to don't display the price for new customized applications, but what about approved/finalized ones? I mean, at some point the `sponsorship_fee` will have the real and final agreed amount even for customized applications. So, what are these conditions?

**Comment by ewdurbin** [comment]
> @ewdurbin I updated this PR with 2 new commits addressing your comments. About the buttons style, I didn't commit a few lines in the final `style.css` file ([e0ddfbf](https://github.com/python/pythondotorg/commit/e0ddfbf07bc15a89c00c78f3532171df38615dd8)) and now it should work for you.
> 
> About the sponsorship fee information, I added a condition ([d1c70dd](https://github.com/python/pythondotorg/commit/d1c70dd6046a347a653215d86c18de4263eb7bdd)) to check if the price should be displayed, but I'm wondering if the condition isn't to restrictive. Because it makes sense to me to don't display the price for new customized applications, but what about approved/finalized ones? I mean, at some point the `sponsorship_fee` will have the real and final agreed amount even for customized applications. So, what are these conditions?

I agree, that is a bit too restrictive. We basically shouldn't show a price for a modified package until approved.

**Review by ewdurbin — APPROVED** [review]

**Pull request merged**